### PR TITLE
Allow user to set bazelOptions.disableStrictDeps

### PR DESCRIPTION
**Pull request by alexeagle**
At Google, all code is required to be strict-deps compliant, except third-party code outside our control.
Users may have a similar need to opt-out some of their code, or might decide not to use this feature on
first-party code either.

Also in this commit:
- Add missing unit test for tsconfig.ts
- Include the source-map-support library so node stack traces can have .ts paths
  (also requires a change in rules_nodejs)
- Update clang-format, should (mostly?) match google-internal style
- Add --config=debug options for debugging node programs

See #89

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -17,7 +17,7 @@ workspace(name = "build_bazel_rules_typescript")
 git_repository(
     name = "build_bazel_rules_nodejs",
     remote = "https://github.com/bazelbuild/rules_nodejs",
-    tag = "0.2.2",
+    commit = "1a3a0de",
```
Comment (achew22) [inline WORKSPACE]: Please use a full SHA1

1a3a0dea51c0e53a2871fbfaba3be178a308a7a3

collisions in the reduced SHA1 address space are almost guaranteed as the commit count grows. Thanks :birthday: paradox!

File: BUILD.bazel [hunk 2]
```diff
@@ -30,10 +30,19 @@ gazelle(
...
+        "tsickle",
+        "tsutils",
+        "source-map-support",
+        "protobufjs",
+        "bytebuffer",
+    ] for ext in [
+        "*.js",
+        "*.json",
+        "*.d.ts",
+    ]]),
```
Comment (evmar) [inline BUILD.bazel]: Is this related to this change?

File: .clang-format [hunk 3]
```diff
@@ -1 +1,2 @@
 BasedOnStyle: Google
+ColumnLimit: 100
```
Comment (evmar) [inline .clang-format]: Is this related to this change?

File: package.json [hunk 4]
```diff
@@ -1,23 +1,28 @@
 {
     "name": "@bazel/typescript",
     "description": "Build TypeScript with Bazel",
-    "version": "0.5.0",
-    "keywords": ["typescript", "bazel"],
+    "version": "0.5.3",
```
Comment (evmar) [inline package.json]: Is this related to this change?

File: tools/bazel.rc [hunk 5]
```diff
@@ -1,2 +1,5 @@
 build --strategy=TypeScriptCompile=worker
 test --test_output=errors
+
+# Enable debugging tests with --config=debug
```
Comment (evmar) [inline tools/bazel.rc]: Is this related to this change?

**Comment by alexeagle** [comment]
I mentioned all the stuff in the commit message, so it was intentional? I can sync into separate CLs.

File: .clang-format (hunk 3 above)
Comment (alexeagle) [reply .clang-format]: removed

File: BUILD.bazel (hunk 2 above)
Comment (alexeagle) [reply BUILD.bazel]: I was getting errors on Mac with exceeding max open file descriptors.
I'll make a separate change for this, or even better, figure out why bazel/ibazel have so many open FDs...

File: WORKSPACE (hunk 1 above)
Comment (alexeagle) [reply WORKSPACE]: don't need to include this in the PR, sorry for so much noise here. Still trying to figure out how to do CI

File: tools/bazel.rc (hunk 5 above)
Comment (alexeagle) [reply tools/bazel.rc]: moved to separate CL

File: package.json (hunk 4 above)
Comment (alexeagle) [reply package.json]: no, can be separate also. Sorry this was so messy.
